lib/src/git.rs

```diff
@@ -1905,3 +2232,512 @@ pub fn parse_gitmodules(
         .collect();
     Ok(ret)
 }
+
+fn convert_git_fetch_output_to_fetch_result(
+    output: Output,
+    prunes: &mut Vec<String>,
+) -> Result<Output, GitFetchError> {
+    // None means signal termination:
+    // 128 is the base for the signal exit codes (128 + signo)
+    let code = output.status.code();
+    if code == Some(0) {
+        return Ok(output);
+    }
+
+    let lossy_err = String::from_utf8_lossy(&output.stderr).to_string();
+    let stderr = String::from_utf8(output.stderr.clone()).map_err(|e| {
+        GitFetchError::ExternalGitError(format!(
+            "external git program failed with non-utf8 output: {e:?}\n--- stderr ---\n{lossy_err}",
+        ))
+    })?;
+
+    // There are some git errors we want to parse out
+
+    // GitFetchError::NoSuchRemote
+    //
+    // To say this, git prints out a lot of things, but the first line is of the
+    // form:
+    // `fatal: '<branch_name>' does not appear to be a git repository`
+    let git_err = {
+        let mut git_err = None;
+        if let Some(first_line) = stderr.lines().next() {
+            if first_line.starts_with("fatal: '")
+                && first_line.ends_with("' does not appear to be a git repository")
+            {
+                let mut split = first_line.split('\'');
+                split.next(); // ignore prefix
+                let branch_name = split.next();
+                if let Some(bname) = branch_name {
+                    git_err = Some(Err(GitFetchError::NoSuchRemote(bname.to_string())));
+                }
+            }
+        }
+
+        git_err
+    };
+    if let Some(e) = git_err {
+        return e;
```

Comment:
suggestion: It seems like you don't need to assign `git_err` to a variable, and can just return it directly on line 2271?